gensim/test/test_fasttext.py

```diff
@@ -155,16 +154,16 @@ def test_load_fasttext_format(self):
         # vector for oov words are slightly different from original FastText due to discarding unused ngrams
         # obtained using a modified version of ./fasttext print-word-vectors lee_fasttext_new.bin
         expected_vec_oov = [
-            -0.23825,
-            -0.58482,
-            -0.22276,
-            -0.41215,
-            0.91015,
-            -1.6786,
-            -0.26724,
-            0.58818,
-            0.57828,
-            0.75801
+            -0.21929,
```

Comment:
What is the reason for this change? Could it because of the `len(ngrams)` issue mentioned in a comment above?

Reply:
That's really a crucial question, but it's not because of the `len(ngrams)` issue.
The change was honestly simply made because the numbers were pretty similar and I thought that the vectors changed a bit since the new code is far more lenient in assigning a vector to unknown ngrams (i.e. once all buckets are occupied, any ngram will result in a vector, even if it was not in the original corpus).

But it looks like there might a bug in the old code: There are **a lot** more ngram vectors in the loaded model (17004) than there are in the model on disk (2762). This is probably because `wv.vectors_ngram = wv.vectors_ngrams.take(ngram_indices, axis=0)` in `init_ngrams_post_load` will result in a `(num_ngrams_total, ngram_vec_len)` matrix. Shouldn't `vectors_ngram` have a shape of `(num_buckets, ngram_vec_len)`?

Reply:
```
...since the new code is far more lenient in assigning a vector to unknown ngrams (i.e. once all buckets are occupied, any ngram will result in a vector, even if it was not in the original corpus).
```
Ahh right, that makes sense, thanks for explaining.

Re: the number of ngram vectors being greater than `num_buckets` (or the number of vectors on disk) - I see why that might have been happening. With a ngram vocab larger than the number of buckets, a lot of ngrams will be mapped to the same indices. And when `.take` is passed a list that contains multiple occurrences of the same index, the vector at that index is "taken" multiple times.
For example - 
```python
all_vectors = np.array([[0.1, 0.3], [0.3, 0.1]])
taken_vectors = all_vectors.take([0, 1, 0], axis=0)
taken_vectors.shape
>>> (3, 2)
```
So it wouldn't result in incorrect results, but yeah, it'd result in unexpectedly high memory usage (and kind of blowing out the whole idea of keeping memory usage constant even with increasing ngram vocabs, out of the water). Thanks for investigating this and explaining it!

@menshikh-iv we're good to merge from my side